code/game/objects/items/binoculars.dm

```diff
@@ -21,42 +22,62 @@
 	. = ..()
 	AddComponent(/datum/component/two_handed, force_unwielded=8, force_wielded=12)
 
-/obj/item/binoculars/Destroy()
-	listeningTo = null
-	return ..()
-
+/**
+ * Handles zooming the user out when they look through the binoculars.
+ *
+ * Arguments:
+ * - [source][/obj/item]: The source of the signal. Equivalent to src.
+ * - [user][/mob]: The mob that has wielded this pair of binoculars.
+ */
 /obj/item/binoculars/proc/on_wield(obj/item/source, mob/user)
 	SIGNAL_HANDLER
 
-	RegisterSignal(user, COMSIG_MOVABLE_MOVED, .proc/on_walk)
+	RegisterSignal(user, COMSIG_MOB_CLIENT_MOVED, .proc/on_walk)
 	RegisterSignal(user, COMSIG_ATOM_DIR_CHANGE, .proc/rotate)
-	listeningTo = user
 	user.visible_message(span_notice("[user] holds [src] up to [user.p_their()] eyes."), span_notice("You hold [src] up to your eyes."))
 	inhand_icon_state = "binoculars_wielded"
 	user.regenerate_icons()
 	user.client.view_size.zoomOut(zoom_out_amt, zoom_amt, user.dir)
 
-/obj/item/binoculars/proc/rotate(atom/thing, old_dir, new_dir)
-	SIGNAL_HANDLER
-
-	if(ismob(thing))
-		var/mob/lad = thing
-		lad.regenerate_icons()
-		lad.client.view_size.zoomOut(zoom_out_amt, zoom_amt, new_dir)
-
-/obj/item/binoculars/proc/on_walk()
-	SIGNAL_HANDLER
-
-	attack_self(listeningTo) //Yes I have sinned, why do you ask?
-
+/**
+ * Handles zooming the user back in when they lower the binoculars.
+ *
+ * Arguments:
+ * - [source][/obj/item]: The source of the signal. Equivalent to src.
+ * - [user][/mob]: The mob that has unwielded this pair of binoculars.
+ */
 /obj/item/binoculars/proc/on_unwield(obj/item/source, mob/user)
 	SIGNAL_HANDLER
 
-	if(listeningTo)
-		UnregisterSignal(user, COMSIG_MOVABLE_MOVED)
-		UnregisterSignal(user, COMSIG_ATOM_DIR_CHANGE)
-		listeningTo = null
```

Comment:
I find it amusing that the code went through the trouble of saving a ref to the the binoculars had signals attached to and never actually used it.